### DISCUSSION TITLE
twice of func (pc *PeerConnection) OnConnectionStateChange(f func(PeerConnectionState))

### DISCUSSION THREAD

**Discussion by http600** [pion/webrtc#2022] [Q&A] [answered]
I found the event of https://github.com/pion/webrtc/blob/635bfd9cad37eff2d2b7004e303ca4d509b8de17/peerconnectionstate.go#L21 will be triggered twice by
https://github.com/pion/webrtc/blob/635bfd9cad37eff2d2b7004e303ca4d509b8de17/peerconnection.go#L480
is this normal?  If this was normal, I have no more questions then, else I need to continue debugging, thanks

**Comment by Sean-Der** [comment, marked as answer]
Hi @http600 

`PeerConnectionStateConnected` could happen an unlimited amount of times. If your PeerConnection goes to `Disconnect` and then reconnects you will see `PeerConnectionStateConnected` again.

Do you see `PeerConnectionStateConnected` fired when it is already in `PeerConnectionStateConnected`? That sounds like a bug, we should only fire when the state changes.
